<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>CÃ©dric Portaneri, Pierre Alliez</author>
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>859</width>
    <height>673</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CGAL - 2D Alpha Wrap</string>
  </property>
  <widget class="Viewer" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="sizeConstraint">
     <enum>QLayout::SizeConstraint::SetNoConstraint</enum>
    </property>
    <property name="bottomMargin">
     <number>9</number>
    </property>
    <item>
     <widget class="Viewer" name="viewer" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>20</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>10</width>
        <height>10</height>
       </size>
      </property>
      <property name="locale">
       <locale language="English" country="UnitedStates"/>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="stats">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <property name="frameShape">
       <enum>QFrame::Shape::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Raised</enum>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="complexity">
         <property name="text">
          <string>Alpha wrap complexity: # edges</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="steiner">
         <property name="text">
          <string>Steiner points: # % projection - # % intersection</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="gate">
         <property name="text">
          <string>Number of gates traversed: #</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="time">
         <property name="text">
          <string>Total execution time: # s</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>859</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuInput">
    <property name="title">
     <string>Input</string>
    </property>
    <addaction name="actionLoad"/>
    <addaction name="actionSaveInput"/>
    <addaction name="actionClear"/>
    <addaction name="separator"/>
    <addaction name="actionScreenshot"/>
    <addaction name="actionScreenshotSettings"/>
   </widget>
   <widget class="QMenu" name="menuAlgo">
    <property name="title">
     <string>Algo</string>
    </property>
    <addaction name="actionExplodeInput"/>
    <addaction name="actionSmoothInput"/>
    <addaction name="actionAlphaWrap"/>
    <addaction name="actionNextStep"/>
    <addaction name="actionNext10Steps"/>
    <addaction name="actionTerminate"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionViewInsideOutside"/>
    <addaction name="actionViewTriangulationEdge"/>
    <addaction name="actionViewVoronoiDiagram"/>
    <addaction name="actionViewEmptyCircles"/>
    <addaction name="actionViewSteinerPoint"/>
    <addaction name="actionViewNextGate"/>
    <addaction name="actionViewNextGateEmptyCircle"/>
    <addaction name="actionViewInput"/>
    <addaction name="actionViewAlphaWrap"/>
    <addaction name="actionViewStats"/>
   </widget>
   <addaction name="menuInput"/>
   <addaction name="menuAlgo"/>
   <addaction name="menuView"/>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="text">
    <string>Clear</string>
   </property>
   <property name="statusTip">
    <string>Clear</string>
   </property>
   <property name="shortcut">
    <string>Space</string>
   </property>
  </action>
  <action name="actionLoad">
   <property name="text">
    <string>Load...</string>
   </property>
   <property name="statusTip">
    <string>Load PSLG</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionAlphaWrap">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Alpha Wrap</string>
   </property>
   <property name="shortcut">
    <string>Return</string>
   </property>
  </action>
  <action name="actionNextStep">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Next Step (J)</string>
   </property>
   <property name="shortcut">
    <string>J</string>
   </property>
  </action>
  <action name="actionNext10Steps">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Next 10 Steps (K)</string>
   </property>
   <property name="shortcut">
    <string>K</string>
   </property>
  </action>
  <action name="actionTerminate">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Terminate (L)</string>
   </property>
   <property name="shortcut">
    <string>L</string>
   </property>
  </action>
  <action name="actionExplodeInput">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Explode Input</string>
   </property>
   <property name="shortcut">
    <string>E</string>
   </property>
  </action>
  <action name="actionViewInsideOutside">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>View Inside/Outside</string>
   </property>
   <property name="shortcut">
    <string>T</string>
   </property>
  </action>
  <action name="actionViewVoronoiDiagram">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>View Voronoi Diagram</string>
   </property>
   <property name="shortcut">
    <string>Y</string>
   </property>
  </action>
  <action name="actionViewInput">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>View Input</string>
   </property>
   <property name="shortcut">
    <string>I</string>
   </property>
  </action>
  <action name="actionViewAlphaWrap">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>View Alpha Wrap</string>
   </property>
   <property name="shortcut">
    <string>O</string>
   </property>
  </action>
  <action name="actionScreenshot">
   <property name="text">
    <string>Screenshot</string>
   </property>
   <property name="shortcut">
    <string>S</string>
   </property>
  </action>
  <action name="actionScreenshotSettings">
   <property name="text">
    <string>Screenshot Settings</string>
   </property>
  </action>
  <action name="actionViewSteinerPoint">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>View Steiner Point</string>
   </property>
   <property name="shortcut">
    <string>P</string>
   </property>
  </action>
  <action name="actionViewNextGate">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>View Next Gate</string>
   </property>
   <property name="shortcut">
    <string>U</string>
   </property>
  </action>
  <action name="actionViewStats">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>View Stats</string>
   </property>
   <property name="shortcut">
    <string>M</string>
   </property>
  </action>
  <action name="actionSmoothInput">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Smooth Input</string>
   </property>
   <property name="shortcut">
    <string>Z</string>
   </property>
  </action>
  <action name="actionViewEmptyCircles">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>View Empty Circles</string>
   </property>
   <property name="shortcut">
    <string>H</string>
   </property>
  </action>
  <action name="actionViewNextGateEmptyCircle">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>View Next Gate Empty Circle</string>
   </property>
   <property name="shortcut">
    <string>G</string>
   </property>
  </action>
  <action name="actionViewTriangulationEdge">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>View Triangulation Edge</string>
   </property>
   <property name="shortcut">
    <string>D</string>
   </property>
  </action>
  <action name="actionSaveInput">
   <property name="text">
    <string>Save Input...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Viewer</class>
   <extends>QWidget</extends>
   <header>Viewer.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
